src/calendar/calendar-year.tsx

```diff
@@ -0,0 +1,96 @@
+import { useEffect, useState } from "react";
+import dayjs from "dayjs";
+import { CalendarYearProps, VariantYear } from "./types";
+import { YearCell, YearPickerContainer } from "./calendar-year.style";
+
+export const CalendarYear = ({
+    calendarDate,
+    showView,
+    onClick,
+    selectedStartDate,
+}: CalendarYearProps) => {
+    const [yearDate, setYearDate] = useState<string[]>(Array(12).fill(""));
+
+    useEffect(() => {
+        if (showView === "Year") {
+            generateDecadeOfYear();
+        }
+    }, [showView, calendarDate]);
+
+    const generateYearStatus = (date: string) => {
+        const grayYearIndex = [0, 11];
+        const isDecadedYear = grayYearIndex.includes(yearDate.indexOf(date));
+        const year = date.split("-")[0];
+
+        let variant: VariantYear = "default";
+
+        variant = isDecadedYear
+            ? "nextDecaded"
+            : dayjs(selectedStartDate).isSame(date, "year")
+            ? "selectedYear"
+            : dayjs().isSame(date, "year")
+            ? "currentYear"
+            : "default";
+
+        return {
+            year,
+            value: date,
+            variant: variant,
+        };
+    };
+
+    const generateDecadeOfYear = () => {
```

Comment:
The `mm` and `dd` are required for the value user selected. I have not idea how to implement your suggestion into my code. 

Could you share more details?

https://github.com/LifeSG/react-design-system/blob/7f97c403c119ca95662deb94599bff5aa911607f/src/util/calendar-helper.ts#L54

Reply:
yup something like that. use dayjs to either set or add/subtract the years

if you prefer to use add/subtract, i would suggest something like this (pseudo-code)
```
year = calendarDate.year()
decade = Math.floor(yyyy / 10) * 10

base = calendarDate.year(decade)
prev = base.subtract(1, "year")
years = [prev, base]
for (i = 1 to 10) {
  years.push(base.add(i, "year"))
}
```

would prefer not doing `if (i === 0)` etc in loops because that makes it harder to understand the behaviour of each iteration

Reply:
Awesome. Learn a lot from you. Thanks!

Reply:
Great. You logic are awesome. Simple and understandable